### ISSUE TITLE
Install fails on Windows 10

### ISSUE THREAD

**Issue by p-adams** [facebook/reason#868]
I have tried several times to install and run the example project; however, I keep getting this:

`npm WARN addRemoteGit Error: Command failed: git -c core.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
npm WARN addRemoteGit  git+https://github.com/npm-opam/ocaml.git#npm-4.02.3 resetting remote C:\Users\paadams\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-npm-opam-ocaml-git-npm-4-02-3-668766b2 because of error: { Error: Command failed: git -c core.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
npm WARN addRemoteGit   killed: false,
npm WARN addRemoteGit   code: 1,
npm WARN addRemoteGit   signal: null,
npm WARN addRemoteGit   cmd: 'git -c core.longpaths=true config --get remote.origin.url' }`

I either get that error or my terminal just hangs when trying to `npm install`

**Comment by alaingalvan** [comment]
I've been able to get Reason working on Ubuntu bash on Windows with both the npm installation and the opam installation:

![reason-winbash](https://cloud.githubusercontent.com/assets/2251123/20824397/85b2b29a-b82a-11e6-9568-f71b77445fc4.jpg)

Here's a [quick link on how to install it](https://msdn.microsoft.com/en-us/commandline/wsl/install_guide).

Bear in mind, this is effectively just installing Reason on Linux and running it on Windows,

**Comment by OvermindDL1** [comment]
> Bear in mind, this is effectively just installing Reason on Linux and running it on Windows,

Does not help us that run on Windows 2008 or need integration with Windows programs.  ;-)

**Comment by p-adams** [comment]
Yeah. I am unable to update my version of Windows 10. So I cannot take advantage of WinBash, unfortunately.

**Comment by maun** [comment]
I too can install the ExampleProject in the linux subsystem. Is there a way to use the atom from windows with this with autocomplete etc?

**Comment by bobzhang** [comment]
Curious which part of reason depends on m4

**Comment by janicduplessis** [comment]
Looked up the issue a bit and it happens because ubuntu uses dash instead of bash when running sh

Fixed it by running
```
sudo rm /bin/sh
sudo ln -s /bin/bash /bin/sh
```

**Comment by yunxing** [comment]
@bobzhang Just saw this. I think ocamlfind uses m4.

**Comment by bobzhang** [comment]
should not it be a dev dependency only?

**Comment by jordwalke** [comment]
@bobzhang Are you asking about `m4` specifically?

@janicduplessis What do we have to do to make sure everything works out of the box for winbash?

**Comment by bobzhang** [comment]
@jordwalke I am talking about ocamlfind, does reason rely on it for building

**Comment by jordwalke** [comment]
Bob it looks like we depend on it in a couple of unimportant places. I bet we could rip that dependency out! Good find. We are looking to reduce the size of dependencies.
Though I think most people are going to need ocamlfind to build some package they depend on. It's the most commo dependency of all the OCaml dependencies. We can still remove it as a reason dep, but we should also make sure m4 also works. We can package into a sandbox and avoid compiling it if it is already available.

**Comment by chenglou** [comment]
Closing this in favor of the centralized thread at #195. Sorry for all these installations folks...
